src/engine/BoundManager.cpp

```diff
@@ -59,6 +61,12 @@ BoundManager::~BoundManager()
         _tightenedLower[i]->deleteSelf();
         _tightenedUpper[i]->deleteSelf();
     }
+
+    if ( _boundExplainer )
```

Comment:
It seems like bound explainer is deleted by the bound manager, but not always created by it. Is this correct? We generally prefer that creation and deletion happen in the same class. This can get confusing